packages/common/src/store/pages/chat/slice.ts

```diff
@@ -73,7 +73,30 @@ export const chatMessagesAdapter = createEntityAdapter<ChatMessageWithExtras>({
   sortComparer: messageSortComparator
 })
 
-const { selectById: getMessage } = chatMessagesAdapter.getSelectors()
+const {
+  selectById: getMessage,
+  selectEntities: getAllMessages,
+  selectIds: getAllMessageIds
+} = chatMessagesAdapter.getSelectors()
+
+// Recalculate hasTail for the message at index - 1 (the previous message of
+// the message at index).
+const recalculatePreviousMessageHasTail = (
+  chatState: EntityState<ChatMessageWithExtras>,
+  index: number
+): void => {
+  const messageIds = getAllMessageIds(chatState)
+  const chatMessages = getAllMessages(chatState)
+  const prevMessageId = messageIds[index + 1]
+  const prevMessage = chatMessages[prevMessageId]
+  if (!prevMessage) return
+  const newMessageId = messageIds[index]
+  const newMessage = chatMessages[newMessageId]
+  chatMessagesAdapter.updateOne(chatState, {
+    id: prevMessageId,
+    changes: { hasTail: hasTail(prevMessage, newMessage) }
```

Comment:
Is it worth determining if `hasTail` has actually changed before calling `updateOne`? If `updateOne` results in a new state that would cause a re-render, we may not want to do that unless the value has actually changed?

Reply:
So you were right, it actually is an object reassign so it would re-render. However it's only for 1 message at a time which I think is fine. It won't re-render the entire screen.